core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -392,10 +397,36 @@ private[spark] class ExecutorAllocationManager(
   }
 
   /**
-   * Request the cluster manager to remove the given executor.
+   * Request the cluster manager to remove the given executors.
    * Return whether the request is received.
    */
-  private def removeExecutor(executorId: String): Boolean = synchronized {
+  private def removeExecutors(executorIds: Seq[String]): Boolean = synchronized {
+
+    val executorIdsToBeRemoved = executorIds.filter(canBeKilled)
+
+    // Send a request to the backend to kill this executor
+    val removeRequestAcknowledged = testing || client.killExecutors(executorIdsToBeRemoved)
+    if (removeRequestAcknowledged) {
+      val numExistingExecutors = allocationManager.executorIds.size - executorsPendingToRemove.size
+      var index = 0
```

Comment:
This was solely to maintain consistency with earlier log message, which I think is useful and more readable rather than scanning the logs separately to check for the removed executor log and the new desired total.